### PR TITLE
store: add chunk pooling

### PR DESCRIPTION
**Pull request by GiedriusS**
Pool byte slices inside of aggrchunks.

### CODE REVIEW

File: pkg/query/querier.go [hunk 1]
```diff
@@ -133,6 +133,16 @@ type querier struct {
...
+
+	returnChunksMtx sync.Mutex
+	returnChunks    []*storepb.AggrChunk
+}
+
+var returnChunksSlicePool = sync.Pool{
+	New: func() interface{} {
+		r := make([]*storepb.AggrChunk, 0)
+		return &r
+	},
```
Comment (pedro-stanaka) [inline pkg/query/querier.go:145]: Maybe we can use the `pool.BucketedPool` and avoid casts down the road and the need to add a new explicit mutex?

Comment (GiedriusS) [reply pkg/query/querier.go:145]: What min/max sizes and factor would you suggest?

Comment (pedro-stanaka) [reply pkg/query/querier.go:145]: Yeah, you got a point as well. Kinda hard to estimate average chunks touched by queries. Let's leave it like this then.

**Review by pedro-stanaka — APPROVED** [review]
LG

**Pull request merged**